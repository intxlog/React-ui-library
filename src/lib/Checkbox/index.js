import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
//import css
import styles from './styles.module.scss'

const Checkbox = ({
  id,
  classes,
  value,
  name,
  labelText,
  defaultChecked,
  onChange,
  onBlur
}) => {
  return (
    <div
    className={classNames({
      [styles.container]: true,
      [classes] : classes
    })}
  >
      <input
        id={id}
        value={value}
        name={name}
        type={`checkbox`}
        defaultChecked={defaultChecked}
        onChange={onChange}
        onBlur={onBlur}
      />
      <div className={styles.checkbox}></div>
      <label
        htmlFor={id}
      >{labelText}</label>
    </div>
  )
}

//rules for props being passed in
Checkbox.propTypes = {
  id: PropTypes.string,
  classes: PropTypes.string,
  labelText: PropTypes.string.isRequired,
  defaultChecked: PropTypes.bool,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
}

//maintain the name for documentation purposes
Checkbox.displayName = 'Checkbox'

export default Checkbox