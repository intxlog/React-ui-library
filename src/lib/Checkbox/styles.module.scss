@import './../../styles/helpers.scss';

$heightWidth: 14px;

.container {
  position: relative;

  > label {
    font-size: 16px;
    line-height: 22px;
    cursor: pointer;
    padding-left: 20px;
    display: block;
  } 
  
  > input {
    clip: rect(1px, 1px, 1px, 1px);
    width: 1px;
    height: 1px;
    position: absolute;

    &:checked ~ .checkbox::before {
      display: block;
    } 
  }
  
  > .checkbox {
    background-color: white;
    border-radius: 2px;
    height: $heightWidth;
    width: $heightWidth;
    cursor: pointer;
    border: 1px solid $colorBorder;
    position: absolute;
    top: 4px;
    pointer-events: none;
    // overflow: hidden;

    &:before {
      content: '\2714';
      color: $colorBorder;
      position: absolute;
      top: -8px;
      left: 0px;
      font-size: 20px;
      display: none;
    }
  }
}