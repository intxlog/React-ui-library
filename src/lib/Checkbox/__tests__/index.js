import React from 'react'
import { shallow } from 'enzyme'
import Checkbox from '..'

describe('Checkbox', () => {
  let props = {
    labelText: `example`
  }
  
  //create a shallow wrapper that can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<Checkbox {...props}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})