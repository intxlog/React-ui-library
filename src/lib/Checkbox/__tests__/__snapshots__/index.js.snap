// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="container"
>
  <input
    type="checkbox"
  />
  <div
    className="checkbox"
  />
  <label>
    example
  </label>
</div>
`;
